include/dlaf/util_matrix.h

```diff
@@ -40,5 +40,31 @@ void assert_blocksize_square(const Matrix& matrix, std::string function, std::st
                                 " is not square.");
 }
 
+/// @brief Verify if dlaf::Matrix is on local memory
```

Comment:
IMHO it is not clear, I had to look at the code to understand it.

You are checking if a matrix is not distributed, so it is entirely stored locally and elements are not shared with other ranks, right?